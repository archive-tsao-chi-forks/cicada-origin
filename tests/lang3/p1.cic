z : {
  x : { x : String }
  y : String
  id : (String) -> String
}
z = {
  x = { x = "abc" }
  y = "123"
  id = (x) => x
}

@show z
@show z.id//(z.x) // NOTE should be z.id(z.x.x)
@show z.id//(z.x.x)
@show z.id//(z.y)
