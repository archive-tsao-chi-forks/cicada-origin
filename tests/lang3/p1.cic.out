{
  id = (_) => _
  x = {
    x = "abc"
  }
  y = "123"
}: {
  id : (_: String) -> String
  x : {
    x : String
  }
  y : String
}
(_) => _: (_: String) -> String
(_) => _: (_: String) -> String
(_) => _: (_: String) -> String
